### PR TITLE
feat: collect white, blue and yellow orbs

### PR DESCRIPTION
**Pull request by Dampen59**
Simple logic to search for and click on : 
- White Orbs
- Blue Orbs
- Yellow Orbs

Modified shop logic to display full board before collecting the orbs : 
- Open the shop if closed
- Buy the cards (untouched)
- Close the shop when done
- Collect orbs

Added the "charm" slot button, it will be used in a next feature.

Let me know if this feature should have it's entries in the configuration file.

### CODE REVIEW

File: alune/images.py [hunk 1]
```diff
@@ -258,9 +279,11 @@ class Button:
     augment_two = ClickButton(BoundingBox(516, 120, 770, 520))
     augment_three = ClickButton(BoundingBox(863, 120, 1117, 520))
     augment_one_roll = ClickButton(BoundingBox(266, 602, 323, 659))
-    augment_two_roll = ClickButton(BoundingBox(612, 602, 669, 659))
+    augment_two_roll = ClickButton(BoundingBox(612, 602, 669, 659)) 
     augment_three_roll = ClickButton(BoundingBox(659, 602, 1016, 659))
 
+    charmslot = ClickButton(BoundingBox(1070, 50, 1250, 280))
```
Comment (akshualy) [inline alune/images.py:285]: This does not belong here.

File: alune/images.py [hunk 2]
```diff
@@ -258,9 +279,11 @@ class Button:
     augment_two = ClickButton(BoundingBox(516, 120, 770, 520))
     augment_three = ClickButton(BoundingBox(863, 120, 1117, 520))
     augment_one_roll = ClickButton(BoundingBox(266, 602, 323, 659))
-    augment_two_roll = ClickButton(BoundingBox(612, 602, 669, 659))
+    augment_two_roll = ClickButton(BoundingBox(612, 602, 669, 659))
```
Comment (akshualy) [inline alune/images.py:282]: Accidental spacebar press?

File: main.py [hunk 3]
```diff
@@ -144,6 +144,13 @@ async def buy_from_shop(adb_instance: ADB, config: AluneConfig):
         config: An instance of the alune config to use.
     """
     screenshot = await adb_instance.get_screen()
+
+    # Open shop if closed
```
Comment (akshualy) [inline main.py:148]: Remove useless comments please

File: main.py [hunk 4]
```diff
@@ -144,6 +144,13 @@ async def buy_from_shop(adb_instance: ADB, config: AluneConfig):
...
+    isShopClosed = screen.get_button_on_screen(screenshot, Button.shop_closed)
+    if isShopClosed:
```
Comment (akshualy) [inline main.py:150]: You can inline the variable: `if screen.get_button...`

File: main.py [hunk 5]
```diff
@@ -164,6 +171,13 @@ async def buy_from_shop(adb_instance: ADB, config: AluneConfig):
             break
 
         await asyncio.sleep(0.25)
+    await asyncio.sleep(1)
+
+    # Close the shop, if open
+    isShopOpen = screen.get_button_on_screen(screenshot, Button.shop_open)
+    if isShopOpen:
```
Comment (akshualy) [inline main.py:178]: inline

File: main.py [hunk 6]
```diff
@@ -214,6 +228,21 @@ async def take_game_decision(adb_instance: ADB, config: AluneConfig):
 
     await buy_from_shop(adb_instance, config)
 
+    white_orbs_to_collect = screen.get_button_on_screen(screenshot, Button.white_orb)
```
Comment (akshualy) [inline main.py:231]: There's a way to get all matches of cv2 match template. This solution is fine for now, but looping all matches would save time.

**Review by akshualy — COMMENTED** [review]
From a code perspective, this looks relatively fine.

From a bot perspective, there's a few concerns I have.
1. Is the image match reliable enough? The orbs have quite a bit of variance due to their animations.
2. The bot will be stuck on orbs if there is no space left on the bench and or in the item slots.
3. The bot will have a lot of random garbage on board and on bench, preventing the configured trait from taking any meaningful effect.

2 and 3 can be solved if we have a mechanism of equipping items and selling champions, but I think that needs to be solved first before collecting orbs. What's the opinion of the others?

**Comment by Kyrluckechuck** [comment]
> 2 and 3 can be solved if we have a mechanism of equipping items and selling champions, but I think that needs to be solved first before collecting orbs. What's the opinion of the others?

@akshualy that's a good call -- I think this should be addressed first, ideally

**Comment by Jokiblack** [comment]
It would be great to add the charm now, just buy those cards

**Comment by akshualy** [comment]
Closing this for now, code can be looked up in diffs when needed in the future :)

**Pull request closed without merging**